code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -0,0 +1,134 @@
+	var/strikes = 2 //How many times a revenant can die before dying for good
+	var/revealed = 0 //If the revenant can take damage from normal sources.
+	var/inhibited = 0 //If the revenant's abilities are blocked by a chaplain's power.
+
+/mob/living/simple_animal/revenant/Life()
+	..()
+	if(essence < essence_min)
+		essence = essence_min
+		if(strikes > 0)
+			strikes--
+			src << "<span class='boldannounce'>Your essence has dropped below critical levels. You barely manage to save yourself - [strikes ? "you can't keep this up!" : "next time, it's death."]</span>"
+		else if(strikes <= 0)
+			src << "<span class='warning'><b>NO! No... it's too late, you can feel yourself fading...</b></span>"
+			src.notransform = 1
+			src.revealed = 1
+			src.invisibility = 0
+			playsound(src, 'sound/effects/screech.ogg', 100, 1)
+			src.visible_message("<b>The revenant</b> lets out a waning screech as violet mist swirls around its dissolving body!")
+			src.icon_state = "revenant_draining"
+			sleep(30)
+			src.death()
+	maxHealth = essence * 2
+	if(!revealed)
+		health = maxHealth //Heals to full when not revealed
+	if(essence < essence_regen_cap && essence_regen)
+		essence += 1
+
+/mob/living/simple_animal/hostile/carp/Process_Spacemove(var/movement_dir = 0)
+	return 1 //Mainly to prevent the no-grav effect
+
+/mob/living/simple_animal/revenant/say(message)
+	return 0 //Revenants cannot speak out loud.
+
+/mob/living/simple_animal/revenant/Stat()
+	..()
+	if(statpanel("Status"))
+		stat(null, "Current essence: [essence]E")
+
+/mob/living/simple_animal/revenant/New()
+	..()
+	notransform = 1
+	spawn(50)
+		if(src.mind)
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_harvest
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_transmit
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/aoe_turf/revenant_light
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_life_tap
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_seed_drain
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_mindspike
+			src.mind.spell_list += new /obj/effect/proc_holder/spell/targeted/revenant_mindblast
+			notransform = 0
+		else
+			qdel(src)
+
+/mob/living/simple_animal/revenant/death()
+	..(1)
+	src.invisibility = 0
+	visible_message("<span class='danger'>[src] pulses with an eldritch purple light as its form unwinds into smoke.</span>")
+	ghostize()
+	qdel(src)
+	return
+
+/mob/living/simple_animal/revenant/attackby(obj/item/W, mob/living/user, params)
+	..()
+	if(istype(W, /obj/item/weapon/nullrod))
+		src.visible_message("<b>The revenant</b> screeches and flails!", \
+							"<span class='userdanger'>The null rod invokes agony in you! You feel your essence draining away!</span>")
+		src.essence -= 25 //hella effective
+		if(prob(5))
```

Comment:
This is pretty lame. If the essence runs out the revenant is going to lose a strike anyway.